api/app/signals/apps/my_signals/tests/rest_framework/views/test_my_signals_history.py

```diff
@@ -28,13 +46,218 @@ class TestMySignalsHistoryEndpoint(APITestCase):
+        self.assertEqual(response_json[0]['_signal'], f'{signal.uuid}')
+
+        self.assertEqual(response_json[1]['what'], 'UPDATE_SLA')
+        self.assertEqual(response_json[1]['action'], 'Servicebelofte:')
+        self.assertEqual(response_json[1]['description'], f'{signal.category_assignment.stored_handling_message}')
+        self.assertEqual(response_json[1]['_signal'], f'{signal.uuid}')
+
+        # Filter on UPDATE_STATUS
+        response = self.client.get(f'{self.endpoint}/{signal.uuid}/history', data={'what': 'UPDATE_STATUS'},
+                                   **self.request_headers)
+        self.assertEqual(response.status_code, HTTP_200_OK)
+
+        response_json = response.json()
+        self.assertEqual(len(response_json), 1)
+
+        self.assertEqual(response_json[0]['what'], 'UPDATE_STATUS')
+        self.assertEqual(response_json[0]['action'], 'Status gewijzigd naar: Open')
+        self.assertEqual(response_json[0]['description'], f'{signal.status.text}')
+        self.assertEqual(response_json[0]['_signal'], f'{signal.uuid}')
+
+        # Filter on UPDATE_SLA
+        response = self.client.get(f'{self.endpoint}/{signal.uuid}/history', data={'what': 'UPDATE_SLA'},
+                                   **self.request_headers)
+        self.assertEqual(response.status_code, HTTP_200_OK)
+
+        response_json = response.json()
+        self.assertEqual(len(response_json), 1)
+
+        self.assertEqual(response_json[0]['what'], 'UPDATE_SLA')
+        self.assertEqual(response_json[0]['action'], 'Servicebelofte:')
+        self.assertEqual(response_json[0]['description'], f'{signal.category_assignment.stored_handling_message}')
+        self.assertEqual(response_json[0]['_signal'], f'{signal.uuid}')
+
+    def assert_what_in_data(self, what, data, pos=None):
+        if pos:
+            self.assertEqual(data[pos]['what'], what)
+        else:
+            self.assertIn(what, [item['what'] for item in data])
+
+    def assert_action_in_data(self, action, data, pos=None):
+        if pos:
+            self.assertEqual(data[pos]['action'], action)
+        else:
+            self.assertIn(action, [item['action'] for item in data])
+
+    def assert_description_in_data(self, description, data, pos=None):
+        if pos:
+            self.assertEqual(data[pos]['description'], description)
+        else:
+            self.assertIn(description, [item['description'] for item in data])
+
+    def test_history_signal_flow(self):
+        signal = self._create_signal_with_history()
+
+        response = self.client.get(f'{self.endpoint}/{signal.uuid}/history', data={'ordering': 'created_at'},
+                                   **self.request_headers)
+        self.assertEqual(response.status_code, HTTP_200_OK)
+
+        response_json = response.json()
+        self.assertEqual(len(response_json), 9)
+
+        self.assert_what_in_data('UPDATE_STATUS', response_json, 0)
```

Comment:
You could add assertions about the text here so that it is clearer which of the Status update you want to show up or not.